src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -68,7 +68,12 @@ static TracerProviderSdk()
                 // Callback when Activity is started.
                 ActivityStarted = (activity) =>
                 {
-                    if (!Sdk.SuppressInstrumentation && activity.IsAllDataRequested)
+                    if (!activity.IsAllDataRequested)
+                    {
+                        return;
+                    }
+
+                    if (SuppressInstrumentationScope.TryEnter() == 0)

```

Comment:
Should be `if (SuppressInstrumentationScope.TryEnter() <= 0)` no?

Reply:
It is intentional, I've added some comment to clarify the logic.